lib/vanagon/component.rb

```diff
@@ -290,6 +290,14 @@ def get_build_dir
       end
     end
 
+    def get_dependency_hash
+      obj = @options
+      %i[version name].each do |var|
+        obj[var.to_s] = self.instance_variable_get("@#{var}")
```

Comment:
I actually just remembered why I went this route in the first place. There are some components in some projects which do not have versions/shas associated with them. The code which had been written before still would have failed had either of those existed. That was the reason why I had been iterating over the list of symbols. It should have been if the value was not nil then add it to the hash I'm okay with removing the metaprogramming stuff 
 and handling that in a different way.